src/internal/github/assets.go

```diff
@@ -1,14 +1,51 @@
 package github
 
 import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
 	"regexp"
+	httpInternal "registry-stable/internal/http"
 )
 
 type Platform struct {
 	OS   string
 	Arch string
 }
 
+func DownloadAssetContents(ctx context.Context, downloadURL string) (body io.ReadCloser, err error) {
+	httpClient := httpInternal.GetHttpRetryClient()
+
+	log.Printf("Downloading asset, url: %s", downloadURL)
+	req, reqErr := http.NewRequestWithContext(ctx, http.MethodGet, downloadURL, nil)
+	if reqErr != nil {
+		log.Printf("Failed to create request %s", reqErr)
+		err = fmt.Errorf("failed to create request: %w", reqErr)
```

Comment:
Let's clean this up after (//cc @cam72cam / @kislerdm ), but i dont see any reason here to return named properties, 

we could easily just return `nil, err` here.

Reply:
Done in 8abba6412fe3fb6d26eee823a5d2e25f3b55a626